superset-frontend/src/explore/components/DatasourcePanel.tsx

```diff
@@ -154,13 +155,10 @@ const DataSourcePanel = ({
       setList({ columns, metrics });
       return;
     }
-    const filteredColumns = lists.columns.filter(
-      column => column.column_name.indexOf(value) !== -1,
-    );
-    const filteredMetrics = lists.metrics.filter(
-      metric => metric.metric_name.indexOf(value) !== -1,
-    );
-    setList({ columns: filteredColumns, metrics: filteredMetrics });
+    setList({
+      columns: matchSorter(columns, value, { keys: ['column_name'] }),
+      metrics: matchSorter(metrics, value, { keys: ['metric_name'] }),
```

Comment:
Would we want to search by SQL expression and data type as well?

Reply:
do you mean searching metrics and columns properties, which users can't really see?

Reply:
Yes. But I think they do have a way to see both. For metric definition, there is an info icon tooltip, for data type there is an icon (maybe not that useful than the metric definition, though).